File: estructura_datos/ordenación/insertion_sort/ejemplo.py
def insertion_sort(l):
    for i in range(1, len(l)):
        aux = l[i]
        position = None

        for j in range(i - 1, -1, -1):
            position = j - 1

            if l[j] <= aux:
                break

            l[j + 1] = l[j]

        l[position + 1] = aux

    return l


def run():
    print(insertion_sort([2, 7, 1, 5, 0, 4, 3, 8, -1, 2, 4]))
    print(insertion_sort([2, 7, 1, 4, 3, 5, 0, 8, 2, -1, 2]))
    print(insertion_sort([7, 2, 1, 4, 0, 5, 6, 8, 2]))


if __name__ == "__main__":
    run()

File: estructura_datos/ordenación/insertion_sort/ejemplo.js
const insertionSort = (a) => {
  for (let i = 1; i < a.length; i++) {
    let aux = a[i];
    let j;
    for (j = i - 1; j >= 0; j--) {
      if (a[j] <= aux) break;
      a[j + 1] = a[j];
    }
    a[j + 1] = aux;
  }

  return a;
};

console.log(insertionSort([2, 7, 1, 5, 0, 4, 3, 8, -1, 2, 4]));
console.log(insertionSort([2, 7, 1, 4, 3, 5, 0, 8, 2, -1, 2]));
console.log(insertionSort([7, 2, 1, 4, 0, 5, 6, 8, 2]));
